third_party/envoy-proxy/Makefile

```diff
@@ -0,0 +1,59 @@
+include ../../metadata.mk
+
+PACKAGE_NAME ?= github.com/projectcalico/calico/third_party/envoy-proxy
+
+ENVOY_PROXY_IMAGE ?= envoy-proxy
+BUILD_IMAGES ?= $(ENVOY_PROXY_IMAGE)
+
+THIRD_PARTY_REGISTRY ?= gcr.io/unique-caldron-775/cnx/tigera/third-party
```

Comment:
OSS sem build secret may not have access to our internal gcr registry. Do we want to open this?

Reply:
If needed I'm sure we can copy the envoybinary image to a public registry.  But let's first resolve everything else about this PR and then see if we need that extra step.

Reply:
Based on occurrences of "gcr" in .semaphore/release/hashrelease.yml and .semaphore/release/release.yml, it looks like this will not be an immediate problem because the build does have access to gcr.io.

Reply:
@danudey hashreleases for OSS are already using quay, so I think we should just use quay for this. No reason to put these in a private registry for OSS?

Reply:
Agreed, I'll move the envoybinary image to `quay.io/tigera/envoybinary:v1.31.5`.